lib/active_record_doctor/detectors/missing_unique_indexes.rb

```diff
@@ -66,6 +66,7 @@ def validations_without_indexes
               columns[-1] = "lower(#{columns[-1]})" unless case_sensitive
 
               next if unique_index?(model.table_name, columns)
+              next if model.columns_hash[attribute.to_s]&.type == :citext # citext is case-insensitive by default
```

Comment:
Two questions:

1. Why is `&.` needed here?
2. If we change `&.` to just `.`, will the test suite fail?

Reply:
Hey @gregnavis - It [failed on a few tests](https://github.com/gee-forr/active_record_doctor/actions/runs/7538421459/job/20518920392#step:6:64) so I added it in.

Do you think there's possibly a less inelegant way to handle this? I'm not a super fan of the safe-navigation operator myself, and it's normally indicative of a code smell of some sort, but I didn't see a way around it without making the code really contort itself to avoid it.

Let me know if you've got a preferred way to not use `&.`

Reply:
hey @gregnavis - checking in to see if you're happy with my response on ☝🏻 and if there's any other changes you'd like to see?